src/StackExchange.Redis/ConfigurationOptions.cs

```diff
@@ -296,15 +296,25 @@ private static RemoteCertificateValidationCallback TrustIssuerCallback(X509Certi
         {
             if (issuer == null) throw new ArgumentNullException(nameof(issuer));
 
-            return (object _, X509Certificate? certificate, X509Chain? __, SslPolicyErrors sslPolicyError)
-                => sslPolicyError == SslPolicyErrors.RemoteCertificateChainErrors
-                    && certificate is X509Certificate2 v2
-                    && CheckTrustedIssuer(v2, issuer);
+            return (object _, X509Certificate? certificate, X509Chain? __, SslPolicyErrors sslPolicyError) =>
+            {
+                // If we're already valid, there's nothing further to check
+                if (sslPolicyError == SslPolicyErrors.None)
+                {
+                    return true;
+                }
+                // If we're not valid due to chain errors - check against the trusted issuer
+                return sslPolicyError == SslPolicyErrors.RemoteCertificateChainErrors
```

Comment:
this is a `[Flags]` enum; that's fine *if* we mean "if the *only* error is the SSL chain, then we'll check downstream" - which is indeed "as written" - just want to check our intent (maybe worth a comment?)

Reply:
Good call - added comment and some tests against to make it a bit more obvious there.